source/Plugins/Language/Swift/SwiftFormatters.cpp

```diff
@@ -180,67 +98,109 @@ bool lldb_private::formatters::swift::StringGuts_SummaryProvider(
+    // does. It involves inspecting the variant and rearranging bits
+    // to match the 64-bit representation.
+    static ConstString g__count("_count");
+    static ConstString g__variant("_variant");
+    static ConstString g__discriminator("_discriminator");
+    static ConstString g__flags("_flags");
+
+    auto count_sp = object_sp->GetChildAtNamePath({g__count, g__value});
+    if (!count_sp) return false;
+    uint64_t count = count_sp->GetValueAsUnsigned(0);
+
+    auto discriminator_sp =
+      object_sp->GetChildMemberWithName(g__discriminator, true);
+    if (!discriminator_sp) return false;
+    uint64_t discriminator = discriminator_sp->GetValueAsUnsigned(0);
+    if (discriminator > 0x7F) {
+      // The discriminator only has 7 bits on 32-bit platforms.
+      return false;
+    }
+
+    auto flags_sp = object_sp->GetChildAtNamePath({g__flags, g__value});
+    if (!flags_sp) return false;
+    uint64_t flags = flags_sp->GetValueAsUnsigned(0);
+    if (flags > 0xFFFF) return false;
+
+    auto variant_sp = object_sp->GetChildMemberWithName(g__variant, true);
+    if (!variant_sp) return false;
+
+    llvm::StringRef variantCase = variant_sp->GetValueAsCString();
+
+    ValueObjectSP payload_sp;
+    if (variantCase.startswith("immortal")) {
+      static ConstString g_immortal("immortal");
+      // Set the immortal bit in the discriminator.
+      discriminator |= 0x80;
+      payload_sp = variant_sp->GetChildAtNamePath({g_immortal, g__value});
+    } else if (variantCase.startswith("native")) {
+      static ConstString g_native("native");
+      payload_sp = variant_sp->GetChildMemberWithName(g_native, true);
+    } else if (variantCase.startswith("bridged")) {
+      static ConstString g_bridged("bridged");
+      auto anyobject_sp = variant_sp->GetChildMemberWithName(g_bridged, true);
+      if (!anyobject_sp) return false;
+      payload_sp = anyobject_sp->GetChildAtIndex(0, true); // "instance"
+    } else {
+      return false; // Unknown variant
+    }
+    if (!payload_sp) return false;
+    uint64_t pointerBits = payload_sp->GetValueAsUnsigned(LLDB_INVALID_ADDRESS);
+  
+    if (pointerBits == LLDB_INVALID_ADDRESS) return false;
 
-  ValueObjectSP object, otherBits;
-  bool is64Bit = process->GetAddressByteSize() == 8;
-  if (is64Bit) {
-    object = valobj.GetChildAtNamePath({g_object, g_object});
-    otherBits = valobj.GetChildAtNamePath({g_otherBits, g_value});
+    if ((discriminator & 0xB0) == 0xA0) {
+      raw0 = count | (pointerBits << 32);
+      raw1 = flags | (discriminator << 56);
+    } else {
+      raw0 = count | (flags << 48);
+      raw1 = pointerBits | (discriminator << 56);
+    }
   } else {
-    object = valobj.GetChildAtNamePath({g_object});
-    otherBits = valobj.GetChildAtNamePath({g_otherBits, g_value});
+    return false; // Unsupported arch?
   }
 
-  if (!object || !otherBits)
-    return false;
-
-  // Taken from StringObject.swift:
```

Comment:
I found out this comment describing the layout very useful. Can we have something equivalent in the new world order?

Reply:
Right, the new diagram is at https://github.com/milseman/swift/blob/17f7eca94b4b57bd25ef7f7dc4f6dbf32321fc5b/stdlib/public/core/StringObject.swift#L27

When we finalize, I can either copy-paste it in here, or else link to the formal ABI documentation. We'll have a series of diagrams describing the layout of each component, so it might just make sense to link to that in the end.